Automattic/jetpack
# Add /recommendations/evaluation endpoint handling

## Proposed changes:
* Adds recommendation evaluation endpoint handling and saving it as Jetpack option.
* Improves `useWelcomeBanner()` hook
* Improves `useSimpleMutation()` hook cc @CodeyGuyDylan 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
> This PR requires you to point Jetpack to your sandbox with D153037-code patched

* Create JN site (with “Enable Sandbox Access” option enabled)
* Set Jetpack constant JETPACK__SANDBOX_DOMAIN to your sandbox
* Install Jetpack through Jetpack Beta plugin, with `add/recommendations-evaluation-endpoint` feature branch
* Go to `/wp-admin/admin.php?page=my-jetpack` and connect your site, evaluation form will be displayed
* Open Network tab in Web Inspector
* Select some options and click "See solutions"
* Make sure that /evaluation and /save-evaluation endpoints returned successful responses
![CleanShot 2024-06-25 at 17 10 30@2x](https://github.com/Automattic/jetpack/assets/8419292/41bfea1c-ebaa-4c4e-b616-aebb68674ad5)

* Welcome Flow banner should disappear after evaluation is saved ✨
